PetShare/Services/Interfaces/Pagination/IPaginationService.cs

```diff
@@ -0,0 +1,17 @@
+﻿using PetShare.Models;
+
+namespace PetShare.Services.Interfaces.Pagination
+{
+    public interface IPaginationService
+    {
+        public PaginatedResult<T>? GetPage<T>(IEnumerable<T> collection, PaginationQueryRequest query);
+    }
+
+    public struct PaginatedResult<T>
+    {
+        public IEnumerable<T> items;
+        public int pageSize;
+        public int pageNr;
+        public int totalCount;
+    }
```

Comment:
We could just have a `record` (or class) with properties instead. Exposing public fields is kind of weird. 

Also, if we store DTO objects in `items`, they could be in `IReadOnlyList` so that we can just return the list instead of enumerating it when we map to DTO response.

We also can drop `public` from interface methods, but that doesn't really change anything so it's not important.